src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -718,23 +720,18 @@ class spell_dru_innervate : public AuraScript
 {
     PrepareAuraScript(spell_dru_innervate);
 
-    void CalculateAmount(AuraEffect const* aurEff, int32& amount, bool& /*canBeRecalculated*/)
+    void AfterApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
     {
-        if (!aurEff->GetTotalTicks())
-        {
-            amount = 0;
-            return;
-        }
-
         if (Unit* caster = GetCaster())
-            amount = int32(CalculatePct(caster->GetCreatePowerValue(POWER_MANA), amount) / aurEff->GetTotalTicks());
-        else
-            amount = 0;
+            if (AuraEffect const* innervateR2 = caster->GetAuraEffect(SPELL_DRUID_INNERVATE_RANK_2, EFFECT_0))
+                if (GetTarget() != caster)
```

Comment:
check if the target is the caster before getting the AuraEffect.. there is no need to check first if he has the AuraEffect, its not like you handle anything else there